### PR TITLE
Sailfish update to 0.10.1

### PR DESCRIPTION
**Pull request by ramezrawas**
Kindly verify and authenticate the "sailfish" tool update with the updates in the test-data

cheers :)

### CODE REVIEW

**Comment by bgruening** [comment]
@nsoranzo if you have a minute :)

File: tools/sailfish/sailfish.xml [hunk 1]
```diff
@@ -152,63 +162,144 @@
...
+        <param argument="--gcSizeSamp" type="integer" value="1" optional="True"
+            label="The value by which to down-sample transcripts when representing the GC content"
+            help="Larger values will reduce memory usage, but may decrease the fidelity of bias modeling results."/>
+
+        <param argument="--gcSpeedSamp" type="integer" value="1" optional="True"
+            label="The value at which teh fragment length PMF is down-sampled when evaluating GC fragment bias."
+            help="Larger values speed up effective length correction, but may decrease the fidelity of bias modeling results."/>
+
+        <param argument="--strictIntersect" type="boolean" truevalue="--strictIntersect" falsevalue="" checked="False"
+            label="Modifies how orphans are assigned." help="When this flag is set, if the intersection of the
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: Label is vague, I'd call it just "Strict intersect"

File: tools/sailfish/sailfish.xml [hunk 2]
```diff
@@ -152,63 +162,144 @@
...
+            label="Use Variational Bayesian EM algorithm for optimization" help="Use Variational Bayesian EM algorithm rather
+            than the traditional EM angorithm for optimization"/>
 
-        <param argument="--allowOrphans" type="boolean" truevalue="--allowOrphans" falsevalue="" checked="False"
-            label="Consider orphaned reads as valid hits when performing lightweight-alignment"
-            help="This option will increase sensitivity (allow more reads to map and more transcripts to be detected), but may decrease specificity as orphaned alignments are more likely to be spurious."/>
+        <param argument="--discardOrphans" type="boolean" truevalue="--discardOrphans" falsevalue="" checked="False"
+            label="Discard orphaned reads as valid hits when performing lightweight-alignment"
+            help="This option will increase sensitivity (allow more reads to map and more transcripts to be detected),
+            but may decrease specificity as orphaned alignments are more likely to be spurious."/>
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: label/help is a bit different from the tool help, may want to double check.

File: tools/sailfish/sailfish.xml [hunk 3]
```diff
@@ -152,63 +162,144 @@
...
+            <help>
+                <![CDATA[
+                All hits are considered "Valid".]]>
+            </help>
+        </param>
+        <param argument="--enforceLibCompat" type="boolean" truevalue="--enforceLibCompat" falsevalue="" checked="False"
+                label="Enforces strict library compatibility.">
+            <help>
+                <![CDATA[
+                Fragments that map in a manner other than what is specified by the expected library type will be disccarded,
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: s/disccarded/discarded/

File: tools/sailfish/sailfish.xml [hunk 4]
```diff
@@ -152,63 +162,144 @@
...
+                even if there are no mappings that agree with the expected library ttype.]]>
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: s/ttype/type/

File: tools/sailfish/sailfish.xml [hunk 5]
```diff
@@ -152,63 +162,144 @@
...
+            </help>
+        </param>
+        <param argument="--allowDovetail" type="boolean" truevalue="--allowDovetail" falsevalue="" checked="False"
+                label="Allow paired-end reads from the same fragment to doveteil.">
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: s/doveteil/dovetail/

File: tools/sailfish/sailfish.xml [hunk 6]
```diff
@@ -152,63 +162,144 @@
...
+            <help>
+                <![CDATA[
+                Allow paired-end reads from the same fragment to "doveteil", such that the ends of the mapped reads can extend past each other.]]>
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: s/doveteil/dovetail/

**Comment by nsoranzo** [comment]
@ramezrawas This looks very good, thanks for all the updating work!

Planemo complains about .shed.yml :
```
Missing shed metadata field [homepage_url] for repository
```

File: tools/sailfish/sailfish.xml [hunk 7]
```diff
@@ -73,34 +62,55 @@
                 --mates2 ./mate2.$ext
                 --libType "${single_or_paired.orientation}${single_or_paired.strandedness}"
             #end if
-            --output ./
+            --output ./results
             $biasCorrect
+            #if $gcBiasCorrect:
+                --gcBiasCorrect $gcBiasCorrect
+            #end if
```
Comment (ramezrawas) [inline tools/sailfish/sailfish.xml]: It is an optional parameter, what the point of this substitution ?

Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: It's not optional, it's a boolean.

File: tools/sailfish/sailfish.xml [hunk 8]
```diff
@@ -152,63 +162,144 @@
...
-                    label="Perform bias correction" help=""/>
+            label="Perform sequence-specific bias correction" help=""/>
+
+        <param argument="--gcBiasCorrect" type="boolean" truevalue="--gcBiasCorrect" falsevalue="" checked="False"
+            label="Perform fragment GC bias correction" help=""/>
+
+        <param argument="--dumpEq" type="boolean" truevalue="--dumpEq" falsevalue="" checked="False"
+            label="Dump the equivalence class counts that were computed during quasi-mapping." help=""/>
+
+        <param argument="--gcSizeSamp" type="integer" value="1" optional="True"
```
Comment (ramezrawas) [inline tools/sailfish/sailfish.xml]: no limits were mentioned

File: tools/sailfish/sailfish.xml [hunk 9]
```diff
@@ -73,34 +62,51 @@
...
+            --output ./results
             $biasCorrect
+            $gcBiasCorrect
             --threads "\${GALAXY_SLOTS:-4}"
-
-            #if $fldMean:
-                --fldMean $fldMean
+            $dumpEq
+            #if str($gcSizeSamp):
+                --gcSizeSamp str($gcSizeSamp)
```
Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: The `str()` is needed only on the `#if` line, where 0 would be considered false. In the second line is breaking the final command line.

Sorry, I should have been clearer.

Comment (nsoranzo) [inline tools/sailfish/sailfish.xml]: The correct format is:
```
#if str($gcSizeSamp)
    --gcSizeSamp $gcSizeSamp
#end if
```

Comment (ramezrawas) [inline tools/sailfish/sailfish.xml]: I see

**Comment by ramezrawas** [comment]
@nsoranzo @bgruening I think I have addressed everything, and I would like to thank you both for your careful review and recommendations :)

**Comment by bgruening** [comment]
Great contribution! Thanks a lot @ramezrawas and @nsoranzo!

**Pull request merged**